### ISSUE TITLE
RegexpMultiline: matches inside text blocks

### ISSUE THREAD

**Issue by ksobolew** [checkstyle/checkstyle#11742]
I have read check documentation: https://checkstyle.org/config_xxxxxx.html#NameOfAffectedCheck
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
%  javac scratch_1.java                                          

%  cat config.xml                                                
<?xml version="1.0" ?>
<!DOCTYPE module PUBLIC "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN" "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <!-- Annotations -->
    <!-- Allow suppressing with an annotation -->
    <module name="RegexpMultiline">
        <property name="format" value="[^;]\s\)+\s*[\{;,]?\s*\n" />
        <property name="message" value="Whitespace character before closing parenthesis" />
    </module>
</module>

%  cat scratch_1.java                                            
class Scratch
{
    public static void main(String[] args)
    {
        String s = """
                CREATE TABLE table (
                    col integer
                )
                """;
    }
}

%  RUN_LOCALE="-Duser.language=en -Duser.country=US"
%  java -jar checkstyle-10.3-all.jar -c config.xml scratch_1.java
Starting audit...
[ERROR] /Users/kudivuhadi/Library/Application Support/JetBrains/IntelliJIdea2022.2/scratches/scratch_1.java:8: Whitespace character before closing parenthesis [RegexpMultiline]
Audit done.
Checkstyle ends with 1 errors.
```

---

The pattern in the `RegexpMultiline` configuration is supposed to complain about stray white spaces in Java code, so I would consider the error reported above a false positive, because it points to contents of a text block, which is not Java.

**Comment by strkkk** [comment]
There is no good way to do this. Regex checks treat file as text, not a java code, therefore it is not possible to distinguish text block content from usual code line.

I am wondering why dont you use WhitespaceBeforeCheck for this?
In case you want to analyse whitespaces before closing parens in strings, MatchXpathCheck can do this.

**Comment by romani** [comment]
Try to play with https://checkstyle.org/config_coding.html#IllegalTokenText

**Comment by ksobolew** [comment]
> @ksobolew There is no good way to do this. Regex checks treat file as text, not a java code, therefore it is not possible to distinguish text block content from usual code line.

Right, thanks, it actually makes sense that Regex is not syntax-aware.

> I am wondering why dont you use WhitespaceBeforeCheck for this? In case you want to analyse whitespaces before closing parens in strings, MatchXpathCheck can do this.

I honestly don't know :) I didn't write the original rules, I just try to make them work with Java 17. This rule was added about 4 years ago and maybe there was no better way to do it back then? I'll try to replace it with something better. Thanks!

**Comment by nick-mancuso** [comment]
@ksobolew see https://github.com/checkstyle/checkstyle/issues/11653#issuecomment-1180373074 and https://github.com/checkstyle/checkstyle/issues/11867#issuecomment-1180363780 for a idea of how to suppress these violations
